### DISCUSSION TITLE
2.19.2-1

### DISCUSSION THREAD

**Discussion by bb-qq** [bb-qq/r8152#463] [Announcements]
* The base code has been updated to version 2.19.2 which includes the following changes:
  * Microcode updates, especially major corrections to RTL8157
  * Energy Efficient Ethernet(EEE) Improvements for RTL8157
  * Improved handling of 5 Gbps links for RTL8157

Note:

* The installation will fail the first time. After that, run the following command from the SSH terminal:
   `sudo install -m 4755 -o root -D /var/packages/r8152/target/r8152/spk_su /opt/sbin/spk_su`
* Selecting a proper version is needed depending on your DSM version.
  * DSM 7.2 and above
    * use packages with the suffix `_7.2`.
  * DSM 7.0 and DSM 7.1
    * use packages with the suffix `_7.1`.
  * DSM 6.2.x
    * use packages with the suffix `_6.2`.

<hr /><em>This discussion was created from the release <a href='https://github.com/bb-qq/r8152/releases/tag/2.19.2-1'>2.19.2-1</a>.</em>

**Comment by Moxie-91** [comment]
I know I'm a bit boring and tedious, but can you make drivers for my DS model, which is a 413 with a Freescale P1022 processor - Qoriq?
![DS413](https://github.com/user-attachments/assets/853a8d6b-a9fe-435f-bf0d-c628ab2bf2d0)

**Reply by bb-qq** [reply]
Unfortunately, I cannot create the driver for Qoriq because Realtek does not support for older kernels.

**Comment by juofmgoblue** [comment]
Can't speak to stability yet (just installed), but getting 3.6-3.8Gbps using a brand new Wavlink 5GbE (NWU340G w/ RTL8157) on DS1520+ (7.2.1) w/o jumbo frames enabled.  Was using a TrendNET 2.5GbE for the past year+ reliably, and the extra ~1-1.3Gbps is well worth the $23 it took to upgrade!

**Reply by Sjekke** [reply]
23$ nice, in EU you pay the double for this adapter if you can find them ...

**Reply by juofmgoblue** [reply]
On sale right now for $30 on Amazon for Prime Day (maybe USA only? not sure).  But I ordered another for my wife yesterday, instead from Newegg.com again, for $27.

**Comment by jokinblanco** [comment]
hello!! I made the mistake of putting the ssh command before the first failed installation. So it didnt get installed properly, any help to uninstall completely and do it again correctly?? thanks a lot

**Reply by bb-qq** [reply]
Could you try this procedure?
https://github.com/bb-qq/r8152/wiki/Troubleshooting#uninstall-completely--full-reinstallation

**Reply by WolfAGR** [reply]
Hi bb-qq, what is the command for removing a folder @appconf in SSH? I try -rm but it doesn't work.

**Comment by WolfAGR** [comment]
Hi, 
I have aquestion redarding DS414 CPU MARVELL Armada XP MV78230 with DSM 7.1.1-42962 Update 8. 
I try to install the armada spk for 7.1 but everything I try it wont install.
Can anyone help me please with package should I use? 
Thank you.
WolfAGR

**Reply by WolfAGR** [reply]
Never mind founded the correct one that work for DS414. r8152-armada-2.17.1-2_7.1

**Comment by hlcgermany** [comment]
Any idea which version of the driver I have to use on my DS920+ with DSM 7.2.2 ? 

It doesn't matter which version of the driver i try, it says that the selected package is not supported of my NAS model or compatible with my DSM version. So that's not the message I expected, because it's not really a failed installation?

I got the Club 3D CAC-1420 2.5G adapter.

![grafik](https://github.com/user-attachments/assets/5c5041b5-4451-41bb-be42-8fd627087522)

**Reply by WolfAGR** [reply]
First of all read the manual from bb_qq, your DS920+ CPU model is Geminilake. DS920+ use Geminilake CPU not Apollolake CPU.
This is the correct file name "r8152-geminilake-2.19.2-2_7.2.spk" (without the qoute)  that you must download and install on DS920+.
I use DSM 7.2.2-72806 Update 3 on my own DS92+ with USB to ethernet RTL8157 and install r8152-geminilake-2.19.2_7.2.spk.

![DS920+](https://github.com/user-attachments/assets/3eeaa86e-cce5-4130-8aff-2277e2329b63)

**Reply by WolfAGR** [reply]
![DS920_2](https://github.com/user-attachments/assets/ce991a12-9605-4ded-801b-2eaa7ccb030c)

**Reply by hlcgermany** [reply]
Thank you Sir 👍 

I will try that tomorrow, should work fine I suppose.

**Reply by hlcgermany** [reply]
HI @WolfAGR,

so I got the driver installed successfully
![grafik](https://github.com/user-attachments/assets/1261a734-6d61-4ea5-83c3-013730f7159c)

The LAN3 Port is available. i configured my Static IP and stuff, connected it to my switch but the LAN3 port keep saying it's not connected.

![grafik](https://github.com/user-attachments/assets/6fae45b3-3451-468d-b6a5-539ac5dd94f5)

I should mention that I use VLANs but that shouldn't be any issue in my opinion. But who knows. The native 1GB ports on the 920+ are working fine in my environment of course.

**Reply by WolfAGR** [reply]
Did you reboot the DS920+?

![DS920_3](https://github.com/user-attachments/assets/1d117782-b18d-4e64-ba68-f27f2cd7fe8a)

**Reply by hlcgermany** [reply]
Hi,
yes I did.

LAN 3 was active then, but only 100 Mbits. Do you have your adapter on the front USB or rear? Mine is currently connected to the Front USB of my DS920+.

I check the cable just to be sure thats not the issue. But I dont think that the cable is faulty.

**Reply by WolfAGR** [reply]
In the rear. If sync are at 100Mbits check the switch. My USB use RTL8157 5Gbe yours is RTL8156 2.5Gbe. My Switch is manageable and 10Gbe so I can see in the port setting which port sync @ what speed.

**Reply by hlcgermany** [reply]
Yeah i checked that, i got a UniFi Pro Max 24 at home and of course it is plugged into a 2,5GbE Port. I will plug it into the rear port and test another cable later today.

**Reply by hlcgermany** [reply]
Hi, the rear USB port works fine. 
![grafik](https://github.com/user-attachments/assets/05cdab05-e151-447e-8a9e-5e29e19dfb4e)

Thanks for your help mate!

**Reply by WolfAGR** [reply]
You welcome.

**Comment by assmann-ing** [comment]
Hi there,
I tried to install the version r8152-apollolake-2.19.2-2_7.2 on my Synology DS918+ with OS Version DSM 7.2.2-72806 Update 3
But the installation failed, telling me that the paket could not get installed
Even when I plug in the UBS Adapter it won't work but with OS Version 6.2 ( and the dedcated driver) it did work last year
Have I done somthing wrong, or is the a compability issue?

Thanks ahead for the feedback
Rudi

**Comment by WolfAGR** [comment]
What kind of usb to ehternet do you have?

**Reply by assmann-ing** [reply]
I have a Realtek USB-A 2.5 GbE Family Controller from Logitec called DeLock
When I tested the Controller about 1,5 Years ago with OS 6.2 and your Software it worked right away
I did plug in the Controller in the front USB Port, and I tried to install the software with and without connected Controller
Sorry for my delay but I was 2 days out of town

**Reply by WolfAGR** [reply]
That is not what I'm asking. 
Ok, tel me what Realtek chip set do you have currently. If you are using "Club 3D CAC-1420" is using RTL8156 / RTL8156B and if you are using "wl-nwu340g" https://www.amazon.com/WAVLINK-Ethernet-5000Mbps-Converter-Aluminum/dp/B0DBY23BL9 then the Realtek chip is RTL8157.
Between the quotes is the type of the USB to Ethernet. 
I have two Synology, DS414 with RTL8156 because the driver support Armada CPU and DS920+ with RTL5157 connect at the back and they are working.
How to find hardware ID on a Mac that I don't know I use Windows.

**Comment by assmann-ing** [comment]
Here comes the link where I found the information about upgrading a Synology NAS
https://www.galaxus.de/de/page/synology-nas-upgrade-25-gigabit-mit-usb-lan-adapter-27363

And here comes the information concerning Chip-Set
![image](https://github.com/user-attachments/assets/70c1a1c6-d92d-4490-802b-e04f1892498a)

**Reply by WolfAGR** [reply]
You can try it connected to the back of yours DS918+ after installation don't forget to reboot the nas.

**Reply by WolfAGR** [reply]
Or buy Club 3D CAC-1420

**Comment by assmann-ing** [comment]
Thanks for your fast response

**Comment by assmann-ing** [comment]
I ordered your recomended USB Stick 
But with the exsiting one and reoobt and plug in in the back I got the following Message
![image](https://github.com/user-attachments/assets/57d659c8-4da5-4902-8b5f-8c61e3b4e09b)

**Comment by assmann-ing** [comment]
So where shall I look for the readme

**Reply by Dot-Dash** [reply]
Typically on GitHub, find the readme like this:  

![Readme on GitHub](https://github.com/user-attachments/assets/1f425712-5c53-4318-a4ee-57397471241f)

**Comment by assmann-ing** [comment]
Thanks for the hint
I did the cli command, boot the system and used synosystem to start the service
![image](https://github.com/user-attachments/assets/10490c3c-2511-4efd-8a83-4a031a839b3c)
But I don't see the NIC in the gui and I still see an installation error

**Comment by assmann-ing** [comment]
![image](https://github.com/user-attachments/assets/aa818d88-a326-4f70-b2bf-4bd57ecb6c6f)

**Reply by assmann-ing** [reply]
In the interface list I find to eth devices a bond and ovs but no 3rd Ethernet
